notification.py

```diff
@@ -243,7 +269,37 @@ def perform_action(
         app.notify("Unable to locate notification", "Try again?")
         return False
 
+    def show_actions(self, index: int):
+        global notification_actions
+
+        if gui_actions.showing:
+            gui_actions.hide()
+
+        if index == -1:
+            return
+
+        self.update_notifications()
+
+        if (notification := self[index]) is None:
+            return
+
+        for identifier, group in self.notification_groups():
+            if identifier != notification.identifier:
+                continue
+
+            notification_actions = set(notification.actions.keys())
+
+            frame = group.AXFrame
+            break
+
+        gui_actions.x = frame.left - 300
```

Comment:
PyCharm is warning that `frame` may not be initialized, should we check for None?

Reply:
Good catch, fixed.